src/main/java/uk/gov/hmcts/reform/cpo/controllers/CasePaymentOrdersController.java

```diff
@@ -55,4 +70,58 @@ public CasePaymentOrdersController(CasePaymentOrdersServiceImpl casePaymentOrder
             .build();
         return casePaymentOrdersServiceImpl.getCasePaymentOrders(casePaymentOrderQueryFilter);
     }
+
+    @PutMapping(path = CASE_PAYMENT_ORDERS_PATH, consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Update Case Payment Order", notes = "Updates a case payment order")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ApiResponses({
+        @ApiResponse(
+            code = 202,
+            message = ""
+        ),
+        @ApiResponse(
+            code = 400,
+            message = "One or more of the following reasons:"
+                + "\n1) " + ValidationError.ID_REQUIRED
+                + "\n2) " + ValidationError.ID_INVALID
+                + "\n3) " + ValidationError.EFFECTIVE_FROM_REQUIRED
+                + "\n4) " + ValidationError.CASE_ID_REQUIRED
+                + "\n5) " + ValidationError.CASE_ID_INVALID
+                + "\n6) " + ValidationError.ORDER_REFERENCE_REQUIRED
+                + "\n7) " + ValidationError.ACTION_REQUIRED
+                + "\n8) " + ValidationError.RESPONSIBLE_PARTY_REQUIRED,
+            response = String.class,
+            examples = @Example({
+                @ExampleProperty(
+                    value = "{\n"
+                        + "   \"status\": \"400\",\n"
+                        + "   \"error\": \"Bad Request\",\n"
+                        + "   \"path\": \"" + CASE_PAYMENT_ORDERS_PATH + "\",\n"
+                        + "   \"details\": [ \""  + ValidationError.ID_REQUIRED + "\" ]\n"
+                        + "}",
+                    mediaType = APPLICATION_JSON_VALUE)
+            })
+        ),
+        @ApiResponse(
+            code = 401,
+            message = AuthError.AUTHENTICATION_TOKEN_INVALID
+        ),
```

Comment:
would this result in case the IDAM token is invalid?

Reply:
The S2S and IDAM token verification config and swagger has been 'inherited' from the MCA repository.  The only current planned divergence from this is: 

* The S2S verification steps will be enhanced during [CPO-10 _"Implement microservice based authorisation"_](https://tools.hmcts.net/jira/browse/CPO-10).
* Currently both the Create and Update endpoints will throw a 400 error (`IdAMIdCannotBeRetrievedException`: "Idam ID cannot be retrieved") if the User info cannot be loaded from IDAM, i.e. when the created_by field is populated with Used ID.

As the final required token verification behaviour would be needed across all APIs can I suggest that this behaviour is documented under new ACs in either a new ticket or CPO-10: whichever feels more relevant.

Reply:
Ok then, agreed that we will tackle returning a 400 error in case of invalid IDAM token rather than an Unauthorized 401 error as part of a separate ticket. Either CPO-10 or a dedicated one

Reply:
I guess it's controller by spring security oauth2. To me 401 seems to be correct code for authentication errors ((still client error 4xx series). 
https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/oauth2/core/OAuth2ErrorCodes.html#INVALID_TOKEN